docs/dev/README.md

```diff
@@ -0,0 +1,269 @@
+
+These options are documented below:
+
+| Argument | Description | Type | Default value |
+|---|---|---|---|
+| `--csrf-secure-cookie` | Enable or disable Secure attribute on the CSRF cookie | `bool` | `true` |
+| `--help` | Print help message and exit | `bool` | `false` |
+| `--kube-config` | Path to kube config file, uses in cluster config if empty | `string` | `""` |
+| `--pipelines-namespace` | Namespace where Tekton pipelines is installed (assumes same namespace as dashboard if not set) | `string` | `""` |
+| `--triggers-namespace` | Namespace where Tekton triggers is installed (assumes same namespace as dashboard if not set) | `string` | `""` |
+| `--port` | Dashboard port number | `int` | `8080` |
+| `--read-only` | Enable or disable read only mode | `bool` | `false` |
+| `--web-dir` | Dashboard web resources directory | `string` | `""` |
+| `--logout-url` | If set, enables logout on the frontend and binds the logout button to this url | `string` | `""` |
+| `--namespace` | If set, limits the scope of resources watched to this namespace only | `string` | `""` |
+| `--log-level` | Minimum log level output by the logger | `string` | `"info"` |
+| `--log-format` | Format for log output (json or console) | `string` | `"json"` |
+
+Run `dashboard --help` to show the supported command line arguments and their default value directly from the `dashboard` binary.
+
+**Important note:** using `--namespace` ensures that the dashboard is watching resources in the namespace specified (and drives the frontend).
+It doesn't limit actions that can be performed to this namespace only though. It's important that this flag is used AND that rbac rules are setup accordingly.
+
+## Build YAML manifests
+
+Manifests to be deployed come in multiple flavors and are generated using `kustomize`.
+
+The `kustomize` overlays can be found in the overlays folder, they cover the target platform (plain kube or Openshift) as well as the Dashboard variation (read-only or read-write).
+
+Use one of the following command to get the manifests that suits your need:
+```bash
+# Plain Kube
+kustomize build overlays/dev
+kustomize build --load_restrictor none overlays/dev-locked-down
+
+# OpenShift
+kustomize build overlays/dev-openshift --load_restrictor none
+kustomize build overlays/dev-openshift-locked-down --load_restrictor none
+```
+
+We also provide a version for working with OpenShift image streams:
+```bash
+kustomize build overlays/dev-openshift-imagestream --load_restrictor none
+```
+
+## Build and deploy with kustomize and ko
+
+`kustomize` and `ko` can be used together to build the yaml manifests, build and push the dahsboard docker image, and deploy the results in a cluster.
+
+Note that it doesn't cover the frontend, make sure you [built the frontend](#build-the-frontend) before.
+
+Before `ko` can build and push the dashboard docker image you will have to define the `KO_DOCKER_REPO` environment variable.
+
+```bash
+export KO_DOCKER_REPO='docker.io/myusername'
```

Comment:
alternatively `ko.local` to avoid dealing with auth / network issues for an external repo

Reply:
neveer used `ko.local` this is the default for `ko` if `KO_DOCKER_REPO` is not defined ?

Reply:
no it's the same as doing `ko --local` or `ko -L` https://github.com/google/ko#usage
I use it most of the time, only using docker hub when I need to. Even though our image is small I've found docker hub can often hang for a while before uploading and `ko.local` avoids that problem.

Reply:
i see, fixed